sopht_mpi/utils/mpi_utils_2d.py

```diff
@@ -208,34 +208,41 @@ class MPIFieldIOCommunicator2D:
     """
     Class exclusive for field communication across ranks, initialises data types
     that will be used for scattering global fields and aggregating local fields.
-    Builds dtypes based on field_offset (determined from local memory offset of field)
-    This class wont be seen by the user, rather based on field metadata we determine
-    the properties here.
+    Builds dtypes based on ghost_size (determined from stencil width of the
+    employed kernel). This class wont be seen by the user, rather based on field
+    metadata we determine the properties here.
     """
 
-    def __init__(self, field_offset, mpi_construct):
-        # Use offset to define indices for inner cell (actual data without halo)
-        assert field_offset >= 0, "field offset has to be >= 0"
-        self.field_offset = field_offset
-        if self.field_offset == 0:
+    def __init__(self, ghost_size, mpi_construct):
+        # Use ghost_size to define indices for inner cell (actual data without
+        # halo)
+        if ghost_size < 0:
+            raise ValueError("ghost size has to be >= 0")
```

Comment:
lets check also for `not is instance(ghost_size, int)` and print error accordingly..

Reply:
Fixed in [ac994ec](https://github.com/fankiat/sopht-mpi/pull/7/commits/ac994ecb1d393a64e6da14a3a85142bcc15b74e7)